feat(core): add MFA role to principal roles

Johaney-s opened the pull request:
* MFA role signals that principal is authorized by multifactor authentication
* this is resolved by obtaining info from UserInfoEndpoint
* if the acr value matches the one expected and timestamp is not older than allowed timeout, the role is added to principal
* the role itself will be required by critical operations in perun

xflord commented on perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java:
```diff
@@ -3946,4 +3957,57 @@ private static Map<String, Integer> createMappingOfValues(PerunBean complementar
+			return false;
+		}
+		if (parsedReturnedTimestamp.isBefore(Instant.now())) {
+			log.error("MFA auth timestamp " + returnedTimestamp + " was grater than current time");
```
```suggestion
			log.error("MFA auth timestamp " + returnedTimestamp + " was greater than current time");
```
typo

Johaney-s commented:
changes after testing:
- renamed `userInfoEndpointAcrValue` to `userInfoEndpointMfaAcrValue`, because acr values are returned also by other authentication methods
- only MFA_TIMESTAMP is now added to additional informations if acr value matches mfa acr value, acr itself is not later needed
- it is now stored in principal's additional information so we can see updated value in `getPerunPrincipal` result, before it was only stored on creation and not updated there, separate map was used
- fixed wrong comparison of timestamp < now check
- moved retrieving issuer and access token in Api
- there's use case of delegation that is commented as fixme, we're okay keeping it out of MFA for now

balcirakpeter reviewed (changes requested):
We do not need to explicitly call the `userInfoEndPoint` each time we refresh authorization. We need to retrieve information about MFA only during session refresh. Inside the authorization refresh, we work only with the given timestamp.

Johaney-s commented on perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java:
```diff
@@ -3946,4 +3957,51 @@ private static Map<String, Integer> createMappingOfValues(PerunBean complementar
+
+	/**
+	 * Checks, if principal was authorized by Multi-factor authentication.
+	 * The information is resolved in UserInfoEndpointCall and stored in principal's additional informations.
```
it is consistent with the name of the property in perun principal
